src/EditorFeatures/CSharp/StringCopyPaste/StringCopyPasteCommandHandler.cs

```diff
@@ -121,16 +121,35 @@ public void ExecuteCommand(PasteCommandArgs args, Action nextCommandHandler, Com
             if (documentBeforePaste == null || documentAfterPaste == null)
                 return;
 
+            // See if we can determine the information about the code the user copied from.
+            var copyPasteService = documentBeforePaste.Project.Solution.Workspace.Services.GetService<IStringCopyPasteService>();
+            if (copyPasteService is null)
+                return;
+
+            // If that last thing copied was a line copy (or we can't even figure out what it was), don't do
+            // anything.  There is special handling for line copy/paste that we don't want to interfere with.
+            var lastCopyIsLineCopy = copyPasteService.LastCopyWasLineCopy;
+            if (lastCopyIsLineCopy is true or null)
+                return;
+
             var cancellationToken = executionContext.OperationContext.UserCancellationToken;
 
             // When pasting, only do anything special if the user selections were entirely inside a single string
-            // literal token.  Otherwise, we have a multi-selection across token kinds which will be extremely 
+            // token/expression.  Otherwise, we have a multi-selection across token kinds which will be extremely
             // complex to try to reconcile.
             var stringExpressionBeforePaste = TryGetCompatibleContainingStringExpression(
                 documentBeforePaste, selectionsBeforePaste, cancellationToken);
             if (stringExpressionBeforePaste == null)
                 return;
 
+            // Also ensure that all the changes the editor actually applied were inside a single string
+            // token/expression. If the editor decided to make changes outside of the string, we definitely do not want
+            // to do anything here.
+            var stringExpressionBeforePasteFromChanges = TryGetCompatibleContainingStringExpression(
+                documentBeforePaste, new NormalizedSnapshotSpanCollection(snapshotBeforePaste, snapshotBeforePaste.Version.Changes.Select(c => c.OldSpan)), cancellationToken);
+            if (stringExpressionBeforePaste != stringExpressionBeforePasteFromChanges)
+                return;
```

Comment:
this is an additional sanity check which also fixes the issue.  I would like both though as we both dont' want to do anything 'smart' if the editor decides to make the edits outside the string *and* we don't want to do anything smart (for now) with line-copy-paste as it likely needs some though to make sense of what it means with string literals.